nebula-logger/main/log-management/classes/LogBatchPurger.cls

```diff
@@ -9,12 +9,60 @@
  * @see LogBatchPurgeScheduler
  */
 global with sharing class LogBatchPurger implements Database.Batchable<SObject>, Database.Stateful {
+    private final Boolean isSystemDebuggingEnabled;
+
     private String originalTransactionId;
     private Integer totalProcessedRecords = 0;
 
+    // Database.emptyRecycleBin counts as a DML statement per record
+    private static Integer MAX_RECORDS_DELETED = (Limits.getLimitDmlRows() / 2) - 1;
+    private static Integer DELETED_COUNT = 0;
+
     private class LogBatchPurgerException extends Exception {
     }
 
+    global LogBatchPurger() {
+        this.isSystemDebuggingEnabled = Logger.getUserSettings()?.EnableSystemMessages__c == true;
+    }
+
+    private class LogDeleter implements System.Queueable {
+        private final List<SObject> recordsToDelete;
+        public LogDeleter(List<SObject> recordsToDelete) {
+            this.recordsToDelete = recordsToDelete;
+        }
+
+        public void process() {
+            if (DELETED_COUNT + this.recordsToDelete.size() < MAX_RECORDS_DELETED) {
+                this.hardDelete(this.recordsToDelete);
+                this.recordsToDelete.clear();
+            } else {
```

Comment:
It looks like [in codecov](https://app.codecov.io/gh/jongpie/NebulaLogger/compare/167/diff#diff-bmVidWxhLWxvZ2dlci9tYWluL2xvZy1tYW5hZ2VtZW50L2NsYXNzZXMvTG9nQmF0Y2hQdXJnZXIuY2xz), there's not any code coverage for this `else` statement - thoughts on marking `MAX_RECORDS_DELETED` as `@testVisible` --> adding a test that verifies that the queueable job is enqueued?

Reply:
Iiiinteresting! Funny story - I had a test for this, and the lines covered, but then I removed it. That was a total mistake on my part, I’ll rewrite it tomorrow and include it here.

Reply:
Sounds like a plan! Thanks!